app/components/shared/login-form.vue

```diff
@@ -140,7 +147,13 @@ async function login(event: FormSubmitEvent<Schema>) {
     <USeparator label="ATAU" />
     <p class="text-center">
       Belum memiliki akun?
-      <ULink class="text-primary-600 hover:underline" to="/register"
+      <ULink
+        class="text-primary-600 hover:underline"
+        :to="
+          redirectPath
+            ? `/register?redirect=${encodeURIComponent(redirectPath)}`
+            : '/register'
+        "
```

Comment:
I found this approach is buggy while maintaining existing query like if you in the `/reading-habits?date=05%2F24%2F2025` page. I encourage you to read the documentation, which we can use query object instead: https://ui.nuxt.com/components/link
![image](https://github.com/user-attachments/assets/0c52d2d4-c496-4ea4-a112-cd822f14b78c)